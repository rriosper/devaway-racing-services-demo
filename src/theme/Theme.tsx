import { ThemeProvider } from '@xstyled/styled-components';
import PropTypes, { InferProps } from 'prop-types';
import React from 'react';

import GlobalStyle from './GlobalStyle';
import themeValues from './values';

const themePropTypes = {
  children: PropTypes.node.isRequired,
};

type ThemeProps = InferProps<typeof themePropTypes>;

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={themeValues}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

Theme.propTypes = themePropTypes;

export default Theme;
