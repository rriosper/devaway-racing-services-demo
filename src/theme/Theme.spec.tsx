import { render } from '@testing-library/react';
import { useTheme } from '@xstyled/styled-components';
import React from 'react';

import Theme from './Theme';

const Component: React.FC = () => {
  const theme = useTheme<Record<string, unknown>>();
  return <div>{JSON.stringify(theme)}</div>;
};

test('Theme', () => {
  const { asFragment } = render(
    <Theme>
      <Component />
    </Theme>,
  );

  expect(asFragment()).toMatchSnapshot();
});
