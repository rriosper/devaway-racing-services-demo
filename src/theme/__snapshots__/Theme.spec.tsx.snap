// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme 1`] = `
<DocumentFragment>
  <div>
    {"borders":{"thin":"1px solid","solid":"2px solid","thick":"3px solid","xthick":"5px solid"},"breakpoints":{"xs":0,"sm":576,"md":768,"lg":992,"xl":1200},"colors":{"text":"#000","bg":"#FFF","card":"#f2f2f2","primary":"#ef482d","foreground":"#909090","foregroundBorder":"#d2d2d2","error":"#db6f6f","gold":"#FAFAD2","silver":"#C0C0C0","bronze":"#cc6633"},"fontSizes":{"0":"1rem","1":"1.2rem","2":"1.3rem","3":"1.3rem","4":"1.5rem","5":"1.8rem"},"fontWeights":{"thin":"100","regular":"300","bold":"500"},"radii":{"card":"3px","smooth":"3px","onlyTop":"3px 3px 0 0"},"shadows":{"card":"0px 1px 2px #333333a3;","smooth":"0px 0px 0px 4px rgba(0,0,0,0.3)"},"sizes":{"xxs":"4px","xs":"8px","sm":"16px","md":"32px","lg":"64px","xl":"128px","xxl":"256px"},"space":{"none":"0","xxs":"1px","xs":"4px","sm":"8px","md":"16px","lg":"32px","xl":"64px","xxl":"128px"},"transitions":{"fast":"all 0.25s","regular":"all 0.5s","slow":"all 1s"},"zIndices":{"splash":100}}
  </div>
</DocumentFragment>
`;
